### PR TITLE
add bodies failsafe

### PR DESCRIPTION
**Pull request by alex-shortt**
add check to make sure bodies exist, and if not, remove the subscription

### CODE REVIEW

**Review by isaac-mason — COMMENTED** [review]
It looks like this would be an issue for vehicle subscriptions as well. If we're adding a failsafe for bodies, I think it makes sense to add one for vehicles too.

As an alternative (or addition?) to this PR we could clean up subscriptions in the [`cannon-worker-api` `removeBodies` handler](https://github.com/pmndrs/use-cannon/blob/master/packages/cannon-worker-api/src/worker/index.ts#L47). That could save us from doing this additional check each frame.

File: packages/cannon-worker-api/src/worker/operations/step.ts [hunk 1]
```diff
@@ -34,6 +34,11 @@ export const step = (
     const [uuid, type, target = 'bodies'] = state.subscriptions[id]
 
     const { bodies, vehicles } = state
+    
+    if(!bodies[uuid]) {
```
[the remark this answers is not in the document]
Comment (alex-shortt) [reply packages/cannon-worker-api/src/worker/operations/step.ts:38]: that's what i get for coding directly from github lol

**Comment by alex-shortt** [comment]
updated the code .. i'm going to throw the fix for `removeBodies` in a separate PR since i'm not 100% sure of the syntax and getting this is asap would be ideal

**Comment by bjornstar** [comment]
Sorry I've been busy for a bit. Taking a look at this now and I'd rather fix the root cause. Sure, you won't get an exception anymore but your physics body will stop moving. Could you provide an example of this failing somewhere?

**Comment by alex-shortt** [comment]
if you want to fix the root cause check #411 . i can fix the pr to not include the failsafe if you'd like

**Comment by isaac-mason** [comment]
I think fixing the root cause might be best. With that, we might not need the failsafe?

**Comment by alex-shortt** [comment]
going to fix root cause instead of failsafe

**Pull request closed without merging**